pkg/tls/tls.go

```
@@ -0,0 +1,238 @@
+		if err != nil {
+			return fmt.Errorf("error making output directory for certs: %w", err)
+		}
+	}
+
+	// Next, copy the chia_ca cert/key
+	err := os.WriteFile(path.Join(outDir, "ca", "chia_ca.crt"), chiaCACrtBytes, 0600)
+	if err != nil {
+		return fmt.Errorf("error copying chia_ca.crt: %w", err)
+	}
+	err = os.WriteFile(path.Join(outDir, "ca", "chia_ca.key"), chiaCAKeyBytes, 0600)
+	if err != nil {
+		return fmt.Errorf("error copying chia_ca.key: %w", err)
+	}
+
+	chiaCACert, err := ParsePemCertificate(chiaCACrtBytes)
+	if err != nil {
+		return fmt.Errorf("error parsing chia_ca.crt")
+	}
+
+	chiaCAKey, err := ParsePemKey(chiaCAKeyBytes)
+	if err != nil {
+		return fmt.Errorf("error parsing chia_ca.key")
+	}
+
+	privateCACertBytes, privateCAKeyBytes, err := GenerateNewCA(path.Join(outDir, "ca", "private_ca"))
+	if err != nil {
+		return fmt.Errorf("error creating private ca pair: %w", err)
+	}
+	privateCACert, err := ParsePemCertificate(privateCACertBytes)
+	if err != nil {
+		return fmt.Errorf("error parsing generated private_ca.crt: %w", err)
+	}
+	privateCAKey, err := ParsePemKey(privateCAKeyBytes)
+	if err != nil {
+		return fmt.Errorf("error parsing generated private_ca.key: %w", err)
+	}
+
+	for _, node := range publicNodeNames {
+		_, _, err = GenerateCASignedCert(chiaCACert, chiaCAKey, path.Join(outDir, node, fmt.Sprintf("public_%s", node)))
+		if err != nil {
+			return fmt.Errorf("error generating public pair for %s: %w", node, err)
+		}
+	}
+
+	for _, node := range privateNodeNames {
+		_, _, err = GenerateCASignedCert(privateCACert, privateCAKey, path.Join(outDir, node, fmt.Sprintf("private_%s", node)))
+		if err != nil {
+			return fmt.Errorf("error generating private pair for %s: %w", node, err)
+		}
+	}
+
+	return nil
+}
+
+// ParsePemCertificate parses a certificate
+func ParsePemCertificate(certPem []byte) (*x509.Certificate, error) {
+	// Load CA certificate
+	caCertBlock, _ := pem.Decode(certPem)
```

What's the other return for `pem.Decode`? Does it make sense to not check it?

---

You probably get similar error handling from the if statement just below though, so it _might_ be fine to ignore the error. However it may add context?